lib/srv/desktop/rdp/rdpclient/src/cliprdr.rs

```diff
@@ -248,14 +268,20 @@ impl Client {
         length: u32,
     ) -> RdpResult<Vec<Vec<u8>>> {
         let resp = FormatDataResponsePDU::decode(payload, length)?;
+        let data_len = resp.data.len();
         debug!(
-            "recieved {} bytes of copied data from Windows Desktop: {:?}",
-            resp.data.len(),
-            std::str::from_utf8(resp.data.as_ref()),
+            "recieved {} bytes of copied data from Windows Desktop",
+            data_len,
         );
 
-        (self.on_remote_copy)(resp.data);
+        // trim the null-terminator, if it exists
+        // (but don't worry about CRLF conversion, most non-Windows systems can handle CRLF well enough)
+        let mut data = resp.data;
+        if !data.is_empty() && data[data.len() - 1] == 0x00 {
+            data.truncate(data.len() - 1);
```

Comment:
Shame there's no `trim_right()` for `Vec<u8>`... that said, you can maybe use `.last()` and `.pop()`:

```
if let Some(0x00) = data.last() {
  let _ = data.pop();
}
```

(or trim all with `s/if/while/`)

I don't know if this is really any better than `.truncate()` though, so YMMV.

Reply:
Yeah, I had `.last()` initially but it returns `Option` and made this look a little more complicated.